pkg/kubecost/allocation.go

```diff
@@ -108,6 +110,59 @@ type RawAllocationOnlyData struct {
 	RAMBytesUsageMax float64 `json:"ramByteUsageMax"`
 }
 
+// PVAllocations is a map of Disk Asset Identifiers to the
+// usage of them by an Allocation as recorded in a PVAllocation
+type PVAllocations map[PVKey]*PVAllocation
+
+// Clone creates a deep copy of a PVAllocations
+func (pv *PVAllocations) Clone() PVAllocations {
+	if pv == nil || *pv == nil {
+		return nil
+	}
+	apv := *pv
+	clonePV := PVAllocations{}
+	for k, v := range apv {
+		clonePV[k] = &PVAllocation{
+			ByteHours: v.ByteHours,
+			Cost:      v.Cost,
+		}
```

Comment:
Don't believe that would work. Also... going to ust resolve the comments that are from Sean's PR because they're already in develop. Let me know if you strongly object to that.